sphinx_gallery/tests/test_gen_rst.py

```diff
@@ -1043,24 +1043,33 @@ def test_capture_repr(
 
 
 @pytest.mark.parametrize(
-    "caprepr_gallery, caprepr_file, expected_out",
+    "caprepr_gallery, caprepr_file, caprepr_block, expected_out",
     [
-        pytest.param(tuple(), ("__repr__",), "2", id="() --> repr"),
-        pytest.param(("__repr__",), "()", "", id="repr --> ()"),
+        pytest.param(tuple(), "()", '("__repr__",)', "2", id="() --> () --> repr"),
+        pytest.param(tuple(), ("__repr__",), None, "2", id="() --> repr --> None"),
+        pytest.param(("__repr__",), "()", None, "", id="repr --> () --> None"),
+        pytest.param(tuple(), ("__repr__",), "()", "", id="() --> repr --> ()"),
+        pytest.param(("__repr__",), "()", "()", "", id="repr --> () --> ()"),
     ],
 )
-def test_per_file_capture_repr(
+def test_capture_repr_per_file_and_per_block(
     gallery_conf,
     caprepr_gallery,
     caprepr_file,
+    caprepr_block,
     expected_out,
     req_mpl,
     req_pil,
     script_vars,
 ):
     """Tests that per file capture_repr overrides gallery_conf."""
```

Comment:
Nitpick, let's update this docstring too.

Reply:
Nice catch!

Reply:
```suggestion
    """Tests that per file/block capture_repr overrides gallery_conf."""
```